retry/tripperware.go

```diff
@@ -26,16 +29,32 @@ func Tripperware(opts ...Option) httpwares.Tripperware {
 			if !o.decider(req) && !isEnabled(req.Context()) {
 				return next.RoundTrip(req)
 			}
-			if o.maxRetry == 0 || getBody(req) == nil {
-				// If we are configured to do no retries or the lack of GetBody function doesn't allow for re-reads of
-				// body data.
+			if o.maxRetry == 0 {
 				return next.RoundTrip(req)
 			}
+
+			getBodyFn := getBody(req)
+			if getBodyFn == nil && req.Body != nil {
```

Comment:
why not move this logic into `getBody` itself? It already handles this for go pre-1.8.